### PR TITLE
Fix for rect fill

### PR DESCRIPTION
**Pull request by eltonjude**
Without fix: http://jsfiddle.net/hg902ra2/5/
With fix: http://jsfiddle.net/hg902ra2/6/

Also fixes the readme to correct supported shape type from square to rect

### CODE REVIEW

File: js/annotations.js [hunk 1]
```diff
@@ -515,7 +515,7 @@
 			}
 
 			if (!shape && shapeOptions && inArray(shapeOptions.type, H.ALLOWED_SHAPES) !== -1) {
-				shape = annotation.shape = renderer[options.shape.type]().attr(shapeOptions.params);
+				shape = annotation.shape = shapeOptions.type === 'rect' ? renderer[options.shape.type]().attr(shapeOptions.params) : renderer[options.shape.type](shapeOptions.params);
```
Comment (pawelfus) [inline js/annotations.js]: Only rect after all? Maybe `renderer[options.shape.type](shapeOptions.params).attr(shapeOptions.params)` then? What do you think?

Comment (eltonjude) [inline js/annotations.js]: Without the if-else, console errors are thrown for the circle case.

With your suggestion, the co-ordinate of circle gets messed up sightly.

With if-else:
image


With your suggestion:
image


Where:
        annotations: [{
            xValue: 2,
            yValue: 350,
            shape: {
                type: 'circle',
                params: {
                    r: 20,
                    fill: '#000'
                }
            }
        }]

**Comment by pawelfus** [comment]
Thank you!

**Comment by eltonjude** [comment]
The co-ordinate of circle gets messed up sightly.

Before:
[image: Inline image 2]

After:
[image: Inline image 1]

Where:
        annotations: [{
            xValue: 3,
            yValue: 150,
            shape: {
                type: 'rect',
                params: {
                r: 2,
                    width: 36,
                    height: 2,
                    fill: '#000'
                }
            }
        },{
            xValue: 2,
            yValue: 350,
            shape: {
                type: 'circle',
                params: {
                r: 20,
                    width: 2,
                    height: 36,
                    fill: '#000'
                }
            }
        }]

**Pull request merged**